File: front_end/lib/models/history_item.dart
class HistoryItem {
  final String id;
  final String problem;
  final String date;

  HistoryItem({required this.id, required this.problem, required this.date});

  factory HistoryItem.fromJson(Map<String, dynamic> json) {
    return HistoryItem(
      id: json['id'] ?? '',
      problem: json['problem'] ?? '',
      date: json['date'] ?? '',
    );
  }
} 

File: back_end/models/history_item.py
class HistoryItem:
    def __init__(self, id, problem, date):
        self.id = id
        self.problem = problem
        self.date = date

    def to_dict(self):
        return {
            'id': self.id,
            'problem': self.problem,
            'date': self.date
        } 